src/shell/lisa_gen_buildroot_rootfs

```diff
@@ -0,0 +1,85 @@
+#!/bin/bash
+set -eu
+
+BUILDROOT_URI=git://git.busybox.net/buildroot
+BUILDROOT_VERSION=2018.08
+BUILDROOT_DIR=$LISA_HOME/src/buildroot
+BUILDROOT_PATCHES_DIR=$(dirname $0)/buildroot_patches
+BUILDROOT_CONFIG=$(dirname $0)/buildroot_config
+
+function print_usage {
+	echo "Usage: $0 [options]"
+	echo "	options:"
+	echo "		-p: purge buildroot to force a fresh build"
+	echo "		-h: print this help message"
+}
+
+while getopts "hp" opt
+do
+	case $opt in
+		h)
+			print_usage
+			exit 0
+			;;
+		p)
+			rm -rf $BUILDROOT_DIR
+			exit 0
+			;;
+		?)
+			print_usage
+			exit -1
+			;;
+	esac
+done
+
+# Executed command @$1 if file @$2 doesn't exist
+function do_if {
+	if [ ! -e $2 ]; then
+		$1
+		touch $2
+	fi
+}
+
+function br_clone {
+	git clone $BUILDROOT_URI $BUILDROOT_DIR
+}
+
+function br_checkout_version {
+	pushd $BUILDROOT_DIR > /dev/null
+	git checkout -b $BUILDROOT_VERSION $BUILDROOT_VERSION
+	popd > /dev/null
+}
+
+function br_patch {
+	pushd $BUILDROOT_DIR > /dev/null
+	for patch in $(ls $BUILDROOT_PATCHES_DIR)
```

Comment:
Would benefit from using shell globing to avoid parsing ls output:
 	for patch in "$BUILDROOT_PATCHES_DIR/*"
https://unix.stackexchange.com/questions/128985/why-not-parse-ls

Reply:
I've reworked that part to avoid any confusion